.cbmc-batch/jobs/aws_array_eq/aws_array_eq_harness.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+
+void aws_array_eq_harness() {
+    /* parameters */
+    void *lhs;
+    void *rhs;
+    size_t lhs_len;
+    size_t rhs_len;
+
+    /* assumptions */
+    __CPROVER_assume(lhs_len <= MAX_BUFFER_SIZE);
+    lhs = bounded_malloc(lhs_len);
```

Comment:
@danielsn should I assume here `nondet_bool() ? NULL : bounded_malloc(lhs_len)` ?

Reply:
At that point, just use ```can_fail_malloc```